Show update stack widget on mismatched template/stack revisions

usirin opened the pull request:
This has one issue that unread count update comes after from stack template update and there is no way to wait for that event to happen without listening changes on `computeController`

usirin committed:
sidebar: set updated stack state on close

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

usirin committed:
sidebar: remove duplication

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

usirin committed:
remote/template: add helper to read revision status

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

usirin committed:
sidebar: check if there is a change on revision status on template change

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

gokmen commented on client/app/lib/remote-extensions/stacktemplate.coffee:
```diff
@@ -6,6 +6,9 @@ remote = require('../remote')
 module.exports = class JStackTemplate extends remote.api.JStackTemplate
 
 
+  getRevisionStatus: -> @template.sum
```
`@getAt 'template.sum'`

gokmen commented on client/app/lib/sidebarcontroller.coffee:
```diff
@@ -60,6 +62,29 @@ module.exports = class SidebarController extends kd.Controller
+          template.on 'update', =>
+            debug 'template has an update event', template
+            stacks = computeController.storage.stacks.get().filter (stack) =>
+              stack.baseStackId is template.getId()
```
`stacks = computeController.storage.stacks.query 'baseStackId', template.getId()`

gokmen reviewed (approved):
👍 just minors

gokmen commented:
```
$(npm bin)/coffeelint --quiet .
  ✗ client/app/lib/sidebarcontroller.coffee
     ✗ #73: Unnecessary fat arrow.

✗ Lint! » 1 error and 0 warnings in 1258 files
```

we can merge after coffeelint issue fixed.

usirin committed:
remote/template: use getAt method for revision status

usirin committed:
sidebar/ctrl: use query method for getting stacks

usirin commented:
This pr results in updated stack event popping up each time name of the template is changing for the owner. I will dig this more, adding `wip` tag.

usirin committed:
sidebar/ctrl: fix stack property to compare base sum

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

usirin committed:
sidebar/ctrl: emit change event even if sums are not changed

This happens only when the title of the template is changed, since there is no sum difference between the stack and the template its generated from, change event wasn't triggered w/o this change.

Signed-off-by: Umut Sirin <umutsirin1@gmail.com>

usirin commented:
@gokmen i started not getting events for `_revisionStatus` changes on `JComputeStack` therefore the unread count icon doesn't appear on realtime events, any ideas?

btw, title change event problem i mentioned above is not happening with `nse` 🤘

gokmen commented:
@usirin that might be related with pubnub, it's happening sometimes for the development config which I'm ending up with restarting everything or checking out the pubnub status.
